luaui/Widgets/gui_options.lua

```diff
@@ -2890,16 +2890,19 @@ function init()
 			onchange = function(_, value)
 				local keyFile = keyLayouts.keybindingLayoutFiles[value]
 
-				if not keyFile or keyFile == '' or not VFS.FileExists(keyFile) then
+				if not keyFile or keyFile == '' then
 					return
 				end
 
-				Spring.SetConfigString("KeybindingFile", keyFile)
+				local isCustom = keyLayouts.keybindingPresets["Custom"] == keyFile
 
-				if WG['bar_hotkeys'] and WG['bar_hotkeys'].reloadBindings then
-					WG['bar_hotkeys'].reloadBindings()
+				if isCustom and not VFS.FileExists(keyFile) then
+					Spring.SendCommands("keysave " .. keyFile)
+					Spring.Echo("Preset Custom selected, file saved at: " .. keyFile)
```

Comment:
It occurs to me that maybe these echoes should be localised?

Reply:
I doubt anyone will be jumping at the chance to translate a one-time debug message. Ideally this runs once and then never appears for the user again.

Reply:
It is a very nice-to-have, and the idea with a lot of translations is that if no one ever translates it, it's no biggie, but it's super easy to make the option for anyone who comes along and wants to just add a translation for whatever reason.
Not that I can remember what function to call or which file to edit :smile: